internal/database/database.go

```diff
@@ -0,0 +1,287 @@
+func (dbs *SQLdb) GetFiles(datasetID string) ([]*FileInfo, error) {
+	var (
+		r     []*FileInfo = nil
+		err   error       = nil
+		count int         = 0
+	)
+
+	for count == 0 || (err != nil && count < dbRetryTimes) {
+		r, err = dbs.getFiles(datasetID)
+		count++
+	}
+	return r, err
+}
+
+// getFiles is the actual function performing work for GetFile
+func (dbs *SQLdb) getFiles(datasetID string) ([]*FileInfo, error) {
+	dbs.checkAndReconnectIfNeeded()
+
+	files := []*FileInfo{}
+	db := dbs.DB
+
+	const query = "SELECT a.file_id, dataset_id, display_file_name, file_name, file_size, " +
+		"decrypted_file_size, decrypted_file_checksum, decrypted_file_checksum_type, file_status from " +
+		"local_ega_ebi.file a, local_ega_ebi.file_dataset b WHERE dataset_id = $1 AND a.file_id=b.file_id;"
+
+	rows, err := db.Query(query, datasetID)
+	if rows.Err() != nil || err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	// Iterate rows
+	for rows.Next() {
+
+		// Read rows into struct
+		fi := &FileInfo{}
+		err := rows.Scan(&fi.FileID, &fi.DatasetID, &fi.DisplayFileName, &fi.FileName, &fi.FileSize,
+			&fi.DecryptedFileSize, &fi.DecryptedFileChecksum, &fi.DecryptedFileChecksumType, &fi.Status)
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+
+		// local_ega_ebi.file:file_size is actually the size of the archive file without header
+		// so we need to increase the encrypted file size by the length of the header if the user
+		// downloaded the files in encrypted format. I set it as 124 which seems to be the default
+		// length, but if files can have greater headers, then we can calculate the length with
+		// fd := GetFile() --> len(fd.Header)
+		fi.FileSize = fi.FileSize + 124
+
+		// Add structs to array
+		files = append(files, fi)
+	}
+
+	return files, nil
+}
+
+// CheckDataset checks if dataset name exists
+func (dbs *SQLdb) CheckDataset(dataset string) (bool, error) {
+	var (
+		r     bool  = false
+		err   error = nil
+		count int   = 0
+	)
+
+	for count == 0 || (err != nil && count < dbRetryTimes) {
+		r, err = dbs.checkDataset(dataset)
+		count++
+	}
+	return r, err
+}
+
+// checkDataset is the actual function performing work for CheckDataset
+func (dbs *SQLdb) checkDataset(dataset string) (bool, error) {
+	dbs.checkAndReconnectIfNeeded()
+
+	db := dbs.DB
+	const query = "SELECT DISTINCT dataset_stable_id FROM local_ega_ebi.filedataset WHERE dataset_stable_id = $1"
+
+	var datasetName string
+	if err := db.QueryRow(query, dataset).Scan(&datasetName); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// CheckFilePermission checks if user has permissions to access the dataset the file is a part of
+func (dbs *SQLdb) CheckFilePermission(fileID string) (string, error) {
+	var (
+		r     string = ""
+		err   error  = nil
+		count int    = 0
+	)
+
+	for count == 0 || (err != nil && count < dbRetryTimes) {
+		r, err = dbs.checkFilePermission(fileID)
+		count++
+	}
+	return r, err
+}
```

Comment:
It's a bit confusing that there are two functions with so similar names (checkFilePermission, CheckFilePermission)

Reply:
This design was copied from how `sda-pipeline` implements query functions and exported functions
https://github.com/neicnordic/sda-pipeline/blob/master/internal/database/db.go

Reply:
Right. I think this is a style decision so I leave it up to you. I would have renamed it to something that makes it clear what each of the functions does. At the moment, the names don't uniquely convey their purposes.